amethyst_error/src/lib.rs

```diff
@@ -0,0 +1,422 @@
+//! Contains the `Error` type and company as used by Amethyst.
+//!
+//! **Note:** This type is not intended to be used outside of Amethyst.
+//! If you are integrating a crate from amethyst to use this, it is recommended that you treat this
+//! type as an opaque [`std::error::Error`].
+//!
+//! [`std::error::Error`]: https://doc.rust-lang.org/std/error/trait.Error.html
+
+// Parts copied from failure:
+// https://github.com/rust-lang-nursery/failure
+
+#![warn(missing_docs, rust_2018_idioms, rust_2018_compatibility)]
+
```

Comment:
`#![cfg_attr(not(feature = "std"), no_std)]` should be here to make this compatible with `no_std`.

Reply:
I was going to consider dropping `no_std` completely based on [a comment above](https://github.com/amethyst/amethyst/pull/1220#discussion_r239069562), what would your input be on this?

Reply:
Backtraces are no longer optional!